daffodil-core/src/main/scala/org/apache/daffodil/runtime1/ElementBaseRuntime1Mixin.scala

```diff
@@ -33,6 +33,24 @@ import org.apache.daffodil.dsom.PrimitiveType
 
 trait ElementBaseRuntime1Mixin { self: ElementBase =>
 
+  // initialize cyclic structure
+  requiredEvaluationsIfActivated(dpathElementCompileInfo.initialize)
+
+  // initialize cyclic structure
+  requiredEvaluationsIfActivated(elementRuntimeData.initialize)
+
+  /**
+   * Initialize cyclic structure
+   * This does all type calculators for the simple types of elements.
+   * There are also global simple type defs to initialize, which have
+   * no corresponding element.
+   */
+  requiredEvaluationsIfActivated(optSimpleType.collect{
+    case stb: SimpleTypeDefBase =>
+      if (stb.optTypeCalculator.isDefined)
+        stb.optTypeCalculator.get.initialize()
+  })
+
```

Comment:
Arguably, the above should be on SimpleTypeRuntime1Mixin. A TypeCalculator is really a Runtime1 thing, and it is carried in the SimpleTypeRuntimeData object (I believe).